src/vm/jitinterface.cpp

```diff
@@ -8634,9 +8634,9 @@ CorInfoIntrinsics CEEInfo::getIntrinsicID(CORINFO_METHOD_HANDLE methodHnd,
     else
     {
         MethodTable * pMT = method->GetMethodTable();
-        if (pMT->IsByRefLike() && pMT->GetModule()->IsSystem())
+        if (pMT->GetModule()->IsSystem())
         {
-            if (pMT->HasSameTypeDefAs(g_pByReferenceClass))
+            if (pMT->IsByRefLike() && pMT->HasSameTypeDefAs(g_pByReferenceClass))
```

Comment:
Curious why you changed this (and if we need to worry that `GetModule()` or `IsSystem()` is significantly more expensive that `IsByRefLike()`).  Also, Span and ReadOnlySpan are both "byref-like" in the sense I'm used to that term being used (transitively contains a managed pointer, can't be put in the heap, etc.), are they not `IsByRefLike()` per this check?